fixes #11648 - katello-service: update to start/stop postgresql

bbuckingham opened the pull request.

bbuckingham committed:
fixes #11648 - katello-service: update to start/stop postgresql

stbenjam commented on katello/katello-service:
```diff
@@ -14,6 +14,7 @@ SERVICES = {
   'pulp_workers'          => 20,
   'pulp_celerybeat'       => 20,
   'pulp_resource_manager' => 20,
+  'postgresql'            => 20,
```
I don't _think_ there's anything before this point that needs postgres today, but I think it might be better if we do the databases early to avoid such a problem later.  What do you think about including it at the same level as mongod?

bbuckingham replied:
@stbenjam, That seems reasonable.  I can move it to earlier in the flow.

bbuckingham replied:
@stbenjam, Updated

stbenjam commented:
APJ

ehelms commented:
Will this break upgrades since we stop services and then attempt to run migrations?

stbenjam commented:
Hmm, yea, but we should ensure the status of all the databases anyway, PR: https://github.com/Katello/katello-installer/pull/257

If its going to get cherry-picked somewhere we need to make sure it goes together.

bbuckingham commented:
@stbenjam, Thanks for making the corresponding change to katello-installer!  We can reference both PRs in the BZ to ensure that if the BZ gets pulled in both changes come in.  That said, I do not see it coming in prior to the next 'release'.

stbenjam commented:
@ehelms mentioned on the other PR that maybe we should include an `--only` option on katello-service, instead of the multiple commands we do in the hooks in that other PR.

Do you think you could add it like we have `--exclude` here? If not I can add it when I have time.

The idea would be we could run:
  `katello-service restart --only postgresql,mongod`

bbuckingham commented:
@stbenjam, Sure, I can look in to adding an --only option.  I assume we can do that separate and independent of this PR and https://github.com/Katello/katello-installer/pull/257.  Correct?

stbenjam commented:
It can go here, because we want to use it instead of doing them one-by-one here https://github.com/Katello/katello-installer/pull/257/files#r38475085

bbuckingham commented:
@stbenjam, Sounds good.

bbuckingham committed:
fixes #11648 - katello-service : add --only option

Adding an --only option, so that the user can perform katello-service
actions on only a set of services.

E.g.
katello-service stop --only postgresql,mongod
katello-service status --only postgresql
...

bbuckingham commented:
@stbenjam, @ehelms, added a --only option to katello-service.  I included it as a separate commit, but can squash, if desired.

ehelms commented:
Thanks for collaborating on this! ACK for me

stbenjam commented:
ACK for me too, thanks for catching that @ehelms

bbuckingham commented:
Thanks for the feedback and review guys!  Merging :)
